### PR TITLE
Support Adding Groups for Transfers

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* /bcgov/entity#13820

*Description of changes:*
- Add new Groups to current Owner Groups
- Fix Home Tenancy label
- Refactor useHomeOwners composable
- Add unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by ChasBoucher — COMMENTED** [review]
Looks good.

**Comment by dimak1** [comment]
> Looks good.

Thanks, but I don't see approval checkmark? @ChasBoucher

**Review by doug-lovett — APPROVED** [review]
LGTM.

File: ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnerGroups.vue [hunk 1]
```diff
@@ -107,10 +114,10 @@ export default defineComponent({
           ? required('Select a group for this owner')
           : []
       }),
-      groupFractionalData: find(getMhrRegistrationHomeOwnerGroups.value, { groupId: props.groupId }),
-      fractionalInfo: computed(() => props.fractionalData),
+      groupFractionalData: find(getTransferOrRegistrationHomeOwnerGroups(), { groupId: props.groupId }),
+      // fractionalInfo: computed(() => props.fractionalData),
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnerGroups.vue]: For future?

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue [hunk 2]
```diff
@@ -186,10 +189,15 @@ export default defineComponent({
       const { data } = await fetchMhRegistration(getMhrInformation.value.mhrNumber)
       const currentOwnerGroups = data?.ownerGroups || [] // Safety check. Should always have ownerGroups
       // Create an ID to each individual owner for UI Tracking
+      // TODO: Remove after API updates to include the ID for Owners
       currentOwnerGroups.forEach(ownerGroup => {
+        ownerGroup.isCurrentOwnerGroup = true
         for (const [index, owner] of ownerGroup.owners.entries()) {
           owner.id = ownerGroup.groupId + (index + 1)
+          owner.isCurrentOwner = true
```
Comment (cameron-eyds) [inline ppr-ui/src/views/mhrInformation/MhrInformation.vue]: I'm not sure this property is required? 
As we discussed i'm including `action` properties on anything Added or Removed, to keep it consistent with other scenarios that track added, removed and edited etc.

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue [hunk 3]
```diff
@@ -186,10 +189,15 @@ export default defineComponent({
       const { data } = await fetchMhRegistration(getMhrInformation.value.mhrNumber)
       const currentOwnerGroups = data?.ownerGroups || [] // Safety check. Should always have ownerGroups
       // Create an ID to each individual owner for UI Tracking
+      // TODO: Remove after API updates to include the ID for Owners
       currentOwnerGroups.forEach(ownerGroup => {
+        ownerGroup.isCurrentOwnerGroup = true
```
Comment (cameron-eyds) [inline ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Same for this property, not sure it is required or used here. Perhaps left over from some testing?

**Review by cameron-eyds — APPROVED** [review]
Looks great, nice job. Just a few comments 👍

File: ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnerGroups.vue (hunk 1 above)
Comment (dimak1) [reply ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnerGroups.vue]: Cleaned up.

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Right, this is not needed. Removed.

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 3 above)
Comment (dimak1) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Also not needed. Removed.

**Comment by dimak1** [comment]
@cameron-freshworks thanks for the review. I cleaned up the code.

**Comment by cameron-eyds** [comment]
> @cameron-freshworks thanks for the review. I cleaned up the code.
@dimak1 
Thanks, looks good!

**Pull request merged**